test/preview/components/retrievers/test_in_memory_embedding_retriever.py

```diff
@@ -128,7 +129,7 @@ def test_valid_run(self):
 
         assert "documents" in result
         assert len(result["documents"]) == top_k
-        assert result["documents"][0].embedding == [1.0, 1.0, 1.0, 1.0]
+        assert (result["documents"][0].embedding == [1.0, 1.0, 1.0, 1.0]).all()
```

Comment:
Let's use `assert np.array_equal(result["documents"][0].embedding, [1.0, 1.0, 1.0, 1.0])` instead. The issue with `.all()` for comparisons is that it behavior differs if an array is empty. For example:
`(np.arange(1) == []).all()` results in `True`.